### PR TITLE
Layer update and visibility fix

### PR DESCRIPTION
**Pull request by protetore**
Fix layer componentDidUpdate that was always returning true and fix visibility

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "react-router-dom": "^4.2.2",
     "react-scripts": "1.1.1",
     "react-syntax-highlighter": "^7.0.0",
-    "styled-components": "^3.1.6"
+    "styled-components": "^3.1.6",
+    "lodash.isequal": "^4.5.0"
```
[the remark this answers is not in the document]
Comment (grillorafael) [reply example/package.json]: It is peer dependency. saw now

File: src/components/Layer.js [hunk 2]
```diff
@@ -24,25 +25,25 @@ export class Layer extends React.PureComponent {
 
     componentDidMount() {
         this.layerPromise = wrappedPromise();
-        this.watchStyleChange();
+        this.props.map.on('load', this.createLayer.bind(this));
+        this.props.map.on('style.load', this.createLayer.bind(this));
```
[the remark this answers is not in the document]
Comment (protetore) [reply src/components/Layer.js]: load will only be called once, and never after the first time
then style.load will be responsible to listen to changes o map style redraw the layer if necessary

File: src/components/Layer.js [hunk 3]
```diff
@@ -61,18 +62,16 @@ export class Layer extends React.PureComponent {
...
-    watchStyleChange() {
-        this.listeners["_map_style_changed"] = this.props.map.on('style.load', this.renderLayer.bind(this));
+            this.layer = undefined;
+        }
     }
 
-    renderLayer() {
-        this.cleanup();
-
+    createLayer() {
```
[the remark this answers is not in the document]
Comment (protetore) [reply src/components/Layer.js]: sure, seems the best option

File: src/components/Layer.js [hunk 4]
```diff
@@ -61,18 +62,16 @@ export class Layer extends React.PureComponent {
             const l = this.props.map.getLayer(this.props.id);
             l && this.props.map.removeLayer(this.props.id);
 
-            const s = this.props.map.getSource(this.props.id);
-            s && this.props.map.removeSource(this.props.id);
-        }
-    }
+            if (this.props.options.source) {
```
[the remark this answers is not in the document]
Comment (protetore) [reply src/components/Layer.js]: forgot to check if its an object, if its an object, the source was added by this layer and it's the layer responsibility to handle it, if it's an object it's an external source

**Pull request merged**